src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,206 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.item = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> last;
+    private Node<T> first;
+    private int size = 0;
+
+    private Node<T> node(int index) {
```

Comment:
не информативное название метода.
```suggestion
    private Node<T> getNodeByIndex(int index) {
```